pymc_bart/pgbart.py

```diff
@@ -72,7 +75,11 @@ def __init__(
         value_bart = vars[0]
         self.bart = model.values_to_rvs[value_bart].owner.op
 
-        self.X = self.bart.X
+        if isinstance(self.bart.X, TensorSharedVariable):
+            self.X = self.bart.X.eval()
```

Comment:
Won't things break when a user changes the shape of `X` (e.g., across different calls to `pm.sample`)?

Reply:
Ideally you would create a variable size symbolic expression for whatever you need in the `step` method, instead of trying to eagerly create variables with `numpy` routines

Reply:
You mean to use aesara everywhere in the step method where X is used?

Reply:
Yes. If X is not a constant and you want to support it, that's the only safe thing to do I think

Reply:
Or at least add a warning when you call `.eval()` (actually you can just call `X.data`), saying that the step will use the current data and will not reflect if it is updated.

Reply:
X is assumed to be constant when computing the trees, so no change during sampling. It is only updated to compute out of sample predictions.

Reply:
I was thinking users had to crate their own `PGStep` object in which case they might wrongly try to use it across different model sampling, while changing the contents in between... but I see that most users will not do it!

In that case my only suggestion is to use the `get._value` instead of `eval`